torchao/quantization/sfdp_int8_fx_pass.py

```diff
@@ -0,0 +1,686 @@
+                    g_u8_bs1(),
+                    c(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                ],
+                d_u8,
+                _sfdp_extra_check_int8(aten.div.Tensor),
+            ),
+            (
+                _sfdp_pattern_int8_4,
+                _sfdp_replacement_int8_4,
+                [
+                    g_u8(),
+                    g_u8(),
+                    g_u8(),
+                    c(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                ],
+                d_u8,
+                _sfdp_extra_check_int8(aten.div.Tensor),
+            ),
+            (
+                _sfdp_pattern_int8_4,
+                _sfdp_replacement_int8_4,
+                [
+                    g_u8_bs1(),
+                    g_u8_bs1(),
+                    g_u8_bs1(),
+                    c(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                    zp(),
+                    scale(),
+                ],
+                d_u8,
+                _sfdp_extra_check_int8(aten.div.Tensor),
+            ),
+        ]
+    for pattern, replacement, args, workaround, extra_check in int8_candidates:
+        # XXX: when adding a new pattern, re-run `gen_attention_patterns` so the pattern
+        # gets serialized to a python file and does not require tracing at runtime.
+        assert isinstance(workaround, dict)
+        name = pattern.__name__
+
+        if len(workaround) >= 1:
+            pattern = partialize_and_update_signature(pattern, dropout=0.0)
+            replacement = partialize_and_update_signature(
+                replacement, dropout=0.0
+            )
+            if "dropout" in workaround:
+                del workaround["dropout"] 
+
+        inference_name = name + "_inference"
+        yield inference_name, {
+            "search_fn": pattern,
+            "replace_fn": replacement,
+            "example_inputs": args,
+            "trace_fn": fwd_only,
+            "pass_dicts": patterns,
+            "extra_check": extra_check,
+            "scalar_workaround": workaround,
+        }
+
+
+@functools.lru_cache(None)
+def _sfdp_init_int8():
+    for key, register_replacement_kwargs in _gen_sfdp_patterns_int8():
+        register_replacement(**register_replacement_kwargs)
+    config.joint_custom_pre_pass = patterns.apply
```

Comment:
is this the official API to add new fusion passes to inductor? what if we have multiple fusion passes that we need to add? i.e. we probably want to move all intel quant passes to torchao in the future as well

Reply:
According to https://github.com/pytorch/pytorch/blob/main/torch/_inductor/config.py#L165, I suppose only one fusion pass could be assigned, also the same case for other customized passes. It is better to expand this to a list of passes. Maybe need more comments from @Chillee @eellison.